src/driver/controller.go

```diff
@@ -191,19 +193,15 @@ func (s *ControllerService) ControllerUnpublishVolume(ctx context.Context, req *
 	}
 	volume := &csi.Volume{ID: volumeID}
 
-	serverID, err := parseNodeID(req.NodeId)
-	if err != nil {
-		return nil, status.Error(codes.NotFound, "node not found")
-	}
-	server := &csi.Server{ID: serverID}
-
-	if err := s.volumeService.Detach(ctx, volume, server); err != nil {
+	if err := s.volumeService.Detach(ctx, volume); err != nil {
 		code := codes.Internal
 		switch err {
 		case volumes.ErrVolumeNotFound:
 			code = codes.NotFound
 		case volumes.ErrServerNotFound:
 			code = codes.NotFound
+		case volumes.ErrLockedServer:
```

Comment:
I think this part can be included in a PR about fixing the locked server errors CSI-driver receives from the hcloud.